Asad935477/Recapify-SaaS
# Integrate Next-Auth for user authentication and update various project configurations

# Integrate Next-Auth for user authentication and update various project configurations

### Description:
This PR introduces user authentication using Next-Auth and includes several other important updates and configurations. 

### Changes:
- Integrated Next-Auth for authentication
- Set up providers for Google and GitHub
- Configured API routes for authentication
- Updated environment variables for authentication providers
- Made schema updates to the users table
- Added basic UI components for login/logout
- Updated Prisma schema and generated migration files
- Modified webpack configuration for optimized builds
- Minor CSS adjustments for better UI consistency

### How to test:
1. Ensure the environment variables are correctly set.
2. Run `prisma migrate dev` to apply the new database schema.
3. Start the application using `npm run dev`.
4. Navigate to the login page and test authentication using Google and GitHub providers.
5. Check if the user data is being stored correctly in the database.
6. Verify other changes like UI adjustments and build configurations are working as expected.

### Notes:
- Remember to secure your environment variables and rotate secrets regularly.
- This is an initial integration; further improvements and bug fixes are expected.

### Screenshots:
Please refer to the attached images for visual updates and UI changes.

### Issue Link:
This PR addresses issue #[issue_number] for integrating authentication and improving project setup.

### Checklist:
- [x] Changes have been verified locally

Please review and merge if everything looks good. Feedback is always appreciated!

---

Let me know if there are any other specific details you'd like to add or change!
